wave_lang/kernel/wave/codegen/handlers.py

```diff
@@ -596,7 +605,33 @@ def handle_generic_binary(emitter: WaveEmitter, node: fx.Node):
                     f"rhs={get_custom(op.rhs)}"
                 )
 
-            result = binary_fn(lhs, rhs, emitter.options)
```

Comment:
It's a little tricky for me to track all these conditions to enable scalarize. I think it'd help if we make centralize all the conditions for scalarize together. 

Any thoughts on doing something like:
```
scalarizable_ops = {operator.add, operator.mul, operator.sub}
should_scalarize = emitter.options.scalarize_packed_math and op in scalarizable_op and _near_mma(node)
```
Then, we won't need maybe_scalarize in the templates :)

Reply:
Done (although, I had some annoyances with python lambdas)